api/v1/controllers/generators.js

```diff
@@ -0,0 +1,137 @@
+   */
+  patchGenerator(req, res, next) {
+    res.send({ status: 200, text: 'PATCH a generator' });
+  },
+
+  /**
+   * POST /generators/{key}
+   *
+   * Modifies the generator and sends it back in the response.
+   *
+   * @param {IncomingMessage} req - The request object
+   * @param {ServerResponse} res - The response object
+   * @param {Function} next - The next middleware function in the stack
+   */
+  postGenerator(req, res, next) {
+    res.send({ status: 201, text: 'POST a generator' });
+  },
+
+  /**
+   * PUT /generators/{key}
+   *
+   * Modifies the generator and sends it back in the response.
+   *
+   * @param {IncomingMessage} req - The request object
+   * @param {ServerResponse} res - The response object
+   * @param {Function} next - The next middleware function in the stack
+   */
+  putGenerator(req, res, next) {
+    res.send({ status: 200, text: 'PUT a generator' });
+  },
+
+  /**
+   * GET /generators/{key}/writers
+   *
+   * Retrieves all the writers associated with the generator
+   *
+   * @param {IncomingMessage} req - The request object
+   * @param {ServerResponse} res - The response object
+   * @param {Function} next - The next middleware function in the stack
+   */
+  getGeneratorWriters(req, res, next) {
+    res.send({ status: 200, text: 'Get writers for generator Y' });
+  },
+
+  /**
+   * GET /generators/{key}/writers/userNameOrId
+   *
+   * Determine whether a user is an authorized writer for an generator and
+   * returns the user record if so.
+   *
+   * @param {IncomingMessage} req - The request object
+   * @param {ServerResponse} res - The response object
+   * @param {Function} next - The next middleware function in the stack
+   */
+  getGeneratorWriter(req, res, next) {
+    res.send({ status: 200, text: 'Get writer X for generator Y' });
+  },
+
+  /**
+   * POST /generators/{key}/writers
+   *
+   * Add one or more users to an generator’s list of authorized writers. If
+   * the "enableRedisSampleStore" is turned on add the writers to the generator
+   * stored in redis
+   *
+   * @param {IncomingMessage} req - The request object
+   * @param {ServerResponse} res - The response object
+   * @param {Function} next - The next middleware function in the stack
+   */
+  postGeneratorWriters(req, res, next) {
+    res.send({ status: 201, text: 'Post generator writer' });
+  },
+
+  /**
+   * DELETE /generators/{keys}/writers/userNameOrId
+   *
+   * Deletes a user from an generator’s list of authorized writers. If the
+   * "enableRedisSampleStore" feature is turned on, delete that user from the
+   * authorized list of writers stored in the cache for this generator.
+   *
+   * @param {IncomingMessage} req - The request object
+   * @param {ServerResponse} res - The response object
+   * @param {Function} next - The next middleware function in the stack
+   */
+  deleteGeneratorWriter(req, res, next) {
+    res.send({ status: 204, text: 'Delete generator writer' });
+  },
+}; // exports
```

Comment:
Because the controller changes require adding tons of tests, I separated out the swagger changes from the controller changes, just to keep things moving forward in small manageable chunks.